### PR TITLE
8341935: javac states that -proc:full is the default but the default as of 23 is -proc:none

### PR DESCRIPTION
**Pull request by jddarcy**
Update javac man page is describe current behavior; includes updates discussed in CSR [JDK-8321319](https://bugs.openjdk.org/browse/JDK-8321319) that were not included in the man pages for JDK 23 GA.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8344477](https://bugs.openjdk.org/browse/JDK-8344477) to be approved

### Warning
&nbsp;⚠️ Found leading lowercase letter in issue title for `8341935: javac states that -proc:full is the default but the default as of 23 is -proc:none`

### Issues
 * [JDK-8341935](https://bugs.openjdk.org/browse/JDK-8341935): javac states that -proc:full is the default but the default as of 23 is -proc:none (**Bug** - P3)
 * [JDK-8344477](https://bugs.openjdk.org/browse/JDK-8344477): javac states that -proc:full is the default but the default as of 23 is -proc:none (**CSR**)

### Reviewers
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22227/head:pull/22227` \
`$ git checkout pull/22227`

Update a local copy of the PR: \
`$ git checkout pull/22227` \
`$ git pull https://git.openjdk.org/jdk.git pull/22227/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22227`

View PR using the GUI difftool: \
`$ git pr show -t 22227`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22227.diff">https://git.openjdk.org/jdk/pull/22227.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22227#issuecomment-2484838699)
</details>

### CODE REVIEW

**Review by lahodaj — APPROVED** [review]
Overall looks good, with one nit comment inline.

File: src/jdk.compiler/share/man/javac.md [hunk 1]
```diff
@@ -338,8 +338,11 @@ file system locations may be directories, JAR files or JMOD files.
 :   Controls whether annotation processing and compilation are done.
     `-proc:none` means that compilation takes place without annotation
     processing. `-proc:only` means that only annotation processing is done,
-    without any subsequent compilation. If this option is not used, or
-    `-proc:full` is specified, annotation processing and compilation are done.
+    without any subsequent compilation. and
```
[the remark this answers is not in the document]
Comment (jddarcy) [reply src/jdk.compiler/share/man/javac.md]: Thanks Jan; looking over the whole documentation for `-proc`, decided to format the information as a bulleted list instead. I think that better captures the intent anyway.

**Review by lahodaj — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**